derive/src/compile_bytecode.rs

```diff
@@ -0,0 +1,135 @@
+use super::Diagnostic;
```

Comment:
What does the name `derive` of the package mean? Maybe we could create a seperate package called `python_precompiler` or something like that, which can be used to precompile python code into bytecode.

Reply:
The `_derive` suffix is sort of a standard for proc macro crates because previously the only procedural macros were custom derives. We have non-derive proc macros already, and I was thinking maybe we could rename the crate to `rustpython_macros`. I don't think that proc macros in the same crate as other code is going to happen very soon, so I'd say it's not really temporary.

Reply:
Temporary is a fun word, and it happens often that the temporary solution becomes the final one. Agreed to keep it like this. If derive is the convention, let's stick with that.